### ISSUE TITLE
RFC: Simplify the content of our release bin/ directory

### ISSUE THREAD

**Issue by bmatican** [YugaByte/yugabyte-db#1660]
Was discussing with @m-iancu recently, that our `bin/` directory, in our unpacked tarball, is a bit bloated with a lot of things the user might not care about. I think there's probably 3 categories of items in there:
- definitely required binaries (like the servers, setup tooling and the client shells)
- maybe useful binaries (like debug tools)
- probably useless binaries (like the old artificial load tester?)

Currently, we have the following in our manifest file, to ship as our `bin/` directory:
```
"bin/bulk_load_cleanup.sh",
"bin/bulk_load_helper.sh",
"bin/configure",
"submodules/yugabyte-installation/bin/yb-ctl",
"bin/yb-prof.py",
"$BUILD_ROOT/bin/log-dump",
"$BUILD_ROOT/bin/sst_dump",
"$BUILD_ROOT/bin/yb-admin",
"$BUILD_ROOT/bin/yb-bulk_load",
"$BUILD_ROOT/bin/yb-generate_partitions_main",
"$BUILD_ROOT/bin/yb_load_test_tool",
"$BUILD_ROOT/bin/yb-master",
"$BUILD_ROOT/bin/yb-pbc-dump",
"$BUILD_ROOT/bin/yb-ts-cli",
"$BUILD_ROOT/bin/yb-tserver",
"$BUILD_ROOT/bin/yb-ysck",
"build-support/post_install.sh",
"thirdparty/installed/common/cqlsh/bin/cqlsh",
"thirdparty/installed/common/cqlsh/bin/cqlsh.py",
"thirdparty/installed/common/bin/redis-cli",
"$THIRDPARTY_BUILD_SPECIFIC_DIR/bin/pprof"
```

I'd propose the following:

1. Keep in `bin/`:
```
yb-ctl
yb-admin
yb-master
yb-tserver
post_install.sh
cqlsh
cqlsh.py
redis-cli
```
^ To these, there's also `psql` and `ysqlsh` which are added as symlinks to their `postgres/` dir counterparts.

2. Move to `tools/`:
```
yb-prof.py
log-dump
sst_dump
yb-pbc-dump
yb-ts-cli
pprof
```

Note: @kmuthukk do you think we need both `pprof` and `yb-prof.py`?

2.1. Move to `tools/bulk_load/` (and make the relevant devops changes):
```
bulk_load_cleanup.sh
bulk_load_helper.sh
yb-bulk_load
yb-generate_partitions_main
```

^ Moving these could cause a breakage on any clusters that might try to use this from YW, however I'm not sure if this is even properly glued in right now? cc @Arnav15 @ramkumarvs

If relevant, we can also make a small devops change to go in with the same release, that would check for the old location, else check for the new location.

3. Remove completely:
> yb-ysck
^ do we even use this? cc @kmuthukk 

> yb_load_test_tool
^ This was the old artificial loadtester, pretty sure we don't need this

> configure
I believe we're not using this or referencing it anywhere, we just use `post_install.sh` cc @mbautin 

cc @rkarthik007 @schoudhury for usability feedback

**Comment by kmuthukk** [comment]
* yb-prof.py and pprof to different things; better to keep them both, but ok to move under tools/ directory.

* In bin/ in your list, I don't see psql & ysqlsh. I would imagine they stay in bin/

**Comment by bmatican** [comment]
> ^ To these, there's also psql and ysqlsh which are added as symlinks to their postgres/ dir counterparts.

yep, @kmuthukk, those are symlinked though, but should stay in bin/

**Comment by rkarthik007** [comment]
Note the decision to rename the following binaries since the earlier discussion:
* psql -> ysqlsh
* cqlsh -> ycqlsh
* redis-cli -> yedish

I feel we should keep only what is required in the `bin/` directory (as outlined by @bmatican above) and move the rest to `tools/` as that is the cleanest.

@bmatican also pointed out offline that Yugabyte Platform has a dependency on the bulk_load scripts, as their path is hardcoded to `bin/`. This feature has not really been used (bulk load from platform), so maybe we deprecate that from the UI/Rest API for now. cc @schoudhury
